march_data_collector/src/march_data_collector/cp_calculator.py

```diff
@@ -1,70 +1,114 @@
+        """Base class to calculate capture point for the exoskeleton."""
+        self._tf_buffer = tf_buffer
+        self._foot_link = foot_link
+
+        self.cp_service = rospy.Service('/march/cp_marker_{fl}'.format(fl=foot_link), CapturePointPose,
+                                        lambda msg: self.get_capture_point(msg))
+
+        self._gravity_constant = 9.81
+        self._prev_t = rospy.Time.now()
+        self._delta_t = 0
+
+        self.x = 0
+        self.y = 0
+        self.z = 0
+        self.vx = 0
+        self.vy = 0
+
+        self._center_of_mass_marker = Marker()
+        self._capture_point_marker = Marker()
+
+        self._capture_point_marker.header.frame_id = 'world'
+        self._capture_point_marker.type = self._capture_point_marker.SPHERE
+        self._capture_point_marker.action = self._capture_point_marker.ADD
+        self._capture_point_marker.pose.orientation.w = 1.0
+
+        self._capture_point_marker.color.a = 1.0
+        self._capture_point_marker.color.g = 1.0
+
+        self._capture_point_marker.scale.x = 0.03
+        self._capture_point_marker.scale.y = 0.03
+        self._capture_point_marker.scale.z = 0.03
+
+    @property
+    def center_of_mass_marker(self):
+        """Center of mass property getter."""
+        return self._center_of_mass_marker
+
+    @center_of_mass_marker.setter
+    def center_of_mass_marker(self, updated_center_of_mass):
+        """Center of mass property setter."""
+        if not isinstance(updated_center_of_mass, Marker):
+            raise TypeError('Given center of mass is not of type; Marker')
+
+        current_time = updated_center_of_mass.header.stamp
+        self._delta_t = (current_time - self._prev_t).to_sec()
+
+        if self._delta_t == 0:
+            return
+
+        self.vx = (updated_center_of_mass.pose.position.x - self._center_of_mass_marker.pose.position.x) / self._delta_t
+        self.vy = (updated_center_of_mass.pose.position.y - self._center_of_mass_marker.pose.position.y) / self._delta_t
+
+        self.x = self._center_of_mass_marker.pose.position.x
+        self.y = self._center_of_mass_marker.pose.position.y
+        self.z = self._center_of_mass_marker.pose.position.z
+
+        self._center_of_mass_marker = updated_center_of_mass
+        self._prev_t = updated_center_of_mass.header.stamp
+
+    def _calculate_capture_point(self, duration):
+        """Calculate a future capture point pose using the inverted pendulum and center of mass.
+
+        :param duration:
+            the amount of seconds away from the current time the capture point should be calculated
+        """
+        rospy.logdebug('time diff: {td}'.format(td=str(self._delta_t)))
+        if self._delta_t == 0:
+            rospy.loginfo('Calculating future center of mass for capture point pose.')
+            new_center_of_mass = InvertedPendulum.numeric_solve_to_t(self.x, self.y, self.z, self.vx, self.vy, duration)
```

Comment:
```suggestion
            new_center_of_mass = InvertedPendulum.numeric_solve_to_t(self.x - world_transform.transform.translation.x, self.y - world_transform.transform.translation.y, self.z - world_transform.transform.translation.z, self.vx, self.vy, duration)
```
These coordinates shoulde be relative to the stable foot.